Build failure: ocf-resource-agents

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *ocf-resource-agents*.

### Build log

Relevant:
```
xml.c:2539:9: error: 'xmlRegisterNodeDefault' is deprecated [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Werror=deprecated-declarations8;;]
 2539 |         xmlRegisterNodeDefault(new_private_data);
      |         ^~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[4]: *** [Makefile:1401: libcrmcommon_la-xml.lo] Error 1
make: *** [GNUmakefile:30: build] Error 2
```

([full log](https://hydra.nixos.org/build/267240271/nixlog/5))

### Additional context

Looks like a GCC update deprecated `xmlRegisterNodeDefault`, which is used by _ocf-resource-agents_.
This also causes the _pacemaker_ build to fail, due to the circular dependency.

### Notify maintainers

@ryantm, @astro.

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.7-pvops, NixOS, 24.11 (Velociraptor), 24.11.20240703.9f4128e`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.23.1`
 - nixpkgs: `not found` (I'm using commit `987a15b56865af0b294d0493b968a7e7d193b44f`)

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/misc/logging/pacemaker/default.nix
+++ b/pkgs/misc/logging/pacemaker/default.nix
@@ -78,6 +78,7 @@ stdenv.mkDerivation rec {
 
   env.NIX_CFLAGS_COMPILE = toString (lib.optionals stdenv.cc.isGNU [
     "-Wno-error=strict-prototypes"
+    "-Wno-error=deprecated-declarations"
   ]);
 
   enableParallelBuilding = true;

